### PR TITLE
Add test to verify sync/publish content already in pulp

### PR DESCRIPTION
**Pull request by kersommoura**
Add test to verify how pulp handles sync/publish of content already on
file system and database.

https://pulp.plan.io/issues/4442
closes: #4442

### CODE REVIEW

File: pulp_file/tests/functional/api/test_publish.py [hunk 1]
```diff
@@ -93,3 +94,50 @@ def test_all(self):
...
+    3. Sync/publish repository A again with download policy immediate.
+    4. No failure in 2 shows that sync went fine when content was
+       not present on the disk and in the database.
+    5. No failure in 3 shows that sync went fine when content was already
+       present on the disk and in the database.
+
+    """
+
+    def test_all(self):
+        """Test whether."""
```
Comment (rochacbruno) [inline pulp_file/tests/functional/api/test_publish.py]: What about a more complete docstring statement?

Comment (kersommoura) [reply pulp_file/tests/functional/api/test_publish.py]: @rochacbruno, I missed that one. Thanks.

**Comment by dralley** [comment]
@kersommoura If this is really a test of underlying Pulp behavior and there's nothing plugin-specific here, do you think it would make sense to move it to pulpcore/tests/functional/api/using_plugin/ ?

**Comment by kersommoura** [comment]
@rochacbruno @dralley I will move this PR to pulpcore. Since it is exercising a pulpcore feature.

**Pull request closed without merging**